barretenberg/cpp/src/barretenberg/commitment_schemes/shplonk/shplemini_verifier.hpp

```diff
@@ -0,0 +1,319 @@
+of a chosing PCS. However, it would be less efficient than ShpleminiVerifier in terms of group and field operations.
+
+
+\subsection Implementation Implementation details
+
+The method \ref accumulate_batch_mul_arguments receives commitments to all prover polynomials, their claimed
+evaluations, the sumcheck challenge, the group element \f$ [1]_1 \f$, and a pointer to the transcript. Its logic could
+be divided into several steps:
+
+1. Receive most of the challenges and prover data.
+2. Run the \ref batch_multivariate_opening_claims method corresponding to step 1 above.
+3. Corresponding to step 2 above:
+   - Run the \ref batch_gemini_claims_received_from_prover method.
+   - Compute the evaluation of the Gemini batched univariate.
+4. Output a triple \f$ (\text{commitments}, \text{scalars}, \text{shplonk_evaluation_point}) \f$ that satiisfies the
+following:
+\f[ \text{batch_mul} (\text{commitments},\ \text{scalars}) = \text{shplonk_opening_claim}.\text{point}
+\f]
+and the sizes of 'commitments' and 'scalars' are equal to
+\f[
+\#\text{claimed_evaluations} + \text{log_circuit_size} + 2
+\f]
+
+The ouput triple is either fed to  the corresponding \ref  bb::KZG< Curve_ >::reduce_verify_shplemini_accumulator
+"KZG method" or \ref  bb::IPA< Curve_ >::reduce_verify_shplemini_accumulator "IPA method". In the case of KZG, we reduce
+\f$ 6 \f$ batch_mul calls needed for the verification of the multivariate evaluation claims to the single batch_mul
+described above. In the case of IPA, the total number of batch_mul calls needed to verify the multivariate evaluation
+claims is reduced by \f$ 5 \f$.
+
+
+\todo Eliminate group operations spent on Shifted Polynomials. See <a
+href="https://github.com/AztecProtocol/barretenberg/issues/1084">the related issue</a>.
+*/
+template <typename Curve> class ShpleminiVerifier_ {
+    using Fr = typename Curve::ScalarField;
+    using GroupElement = typename Curve::Element;
+    using Commitment = typename Curve::AffineElement;
+    using VK = VerifierCommitmentKey<Curve>;
+    using ShplonkVerifier = ShplonkVerifier_<Curve>;
+    using GeminiVerifier = GeminiVerifier_<Curve>;
+
+  public:
+    template <typename Transcript>
+    static ShpleminiAccumulator<Curve> accumulate_batch_mul_arguments(Fr log_N,
+                                                                      RefSpan<Commitment> unshifted_commitments,
+                                                                      RefSpan<Commitment> shifted_commitments,
+                                                                      RefSpan<Fr> claimed_evaluations,
+                                                                      const std::vector<Fr>& multivariate_challenge,
+                                                                      const Commitment& g1_identity,
+                                                                      std::shared_ptr<Transcript>& transcript)
+    {
+        size_t log_circuit_size{ 0 };
```

Comment:
I personally find this style with no line breaks hard to read. Please consider line breaking to group ideas, as you would use paragraphs in an essay.